Public/Src/Cache/ContentStore/Distributed/Utilities/GrpcDistributedPathTransformer.cs

```diff
@@ -42,9 +43,18 @@ public byte[] GetLocalMachineLocation(AbsolutePath cacheRoot)
 
             string networkPathRoot = cacheRoot.Path.Replace(":", "$");
 
-            return
-                Encoding.UTF8.GetBytes(
-                    Path.Combine(@"\\" + _localMachineName, networkPathRoot).ToUpperInvariant());
+            if (OperatingSystemHelper.IsWindowsOS)
+            {
+                // Only unify paths along casing if on Windows
+                networkPathRoot = Path.Combine(@"\\" + _localMachineName, networkPathRoot).ToUpperInvariant();
+            }
+            else
+            {
+                // Path.Combine ignores the first parameter if the second is a rooted path. To get the machine name before the rooted network path, the combination must be done manually.
+                networkPathRoot = @"\\" + _localMachineName + (networkPathRoot.StartsWith(Path.DirectorySeparatorChar.ToString()) ? string.Empty : Path.DirectorySeparatorChar.ToString()) + networkPathRoot;

```

Comment:
>StartsWith [](start = 79, length = 10)

A simpler option which is slightly better on the eyes is the following:

`Path.Combine(@"\\" + _localMachineName + Path.DirectorySeparatorChar, networkPathRoot.TrimLeft(Path.DirectorySeparatorChar.ToString()))`